shimmingtoolbox/shim/sequencer.py

```diff
@@ -23,857 +21,1365 @@
+            coils (ListCoil): List of Coils containing the coil profiles. The coil profiles and the fieldmaps must have
+                          matching units (if fmap is in Hz, the coil profiles must be in hz/unit_shim).
+                          Refer to :class:`shimmingtoolbox.coils.coil.Coil`. Make sure the extent of the coil profiles
+                          are larger than the extent of the fieldmap. This is especially true for dimensions with only
+                          1 voxel(e.g. (50x50x1). Refer to :func:`shimmingtoolbox.shim.sequencer.extend_slice`/
+                          :func:`shimmingtoolbox.shim.sequencer.update_affine_for_ap_slices`
+            method (str): Supported optimizer: 'least_squares', 'pseudo_inverse'. Note: refer to their specific
+                      implementation to know limits of the methods in: :mod:`shimmingtoolbox.optimizer`
+            opt_criteria (str): Criteria for the optimizer 'least_squares'. Supported: 'mse': mean squared error,
+                            'mae': mean absolute error, 'std': standard deviation.
+            mask_dilation_kernel (str): kernel used to dilate the mask. Allowed shapes are: 'sphere', 'cross', 'line'
+                                    'cube'. See :func:`shimmingtoolbox.masking.mask_utils.dilate_binary_mask` for more
+                                    details.
+            mask_dilation_kernel_size (int): Length of a side of the 3d kernel to dilate the mask. Must be odd.
+                                            For example, a kernel of size 3 will dilate the mask by 1 pixel.
+            reg_factor (float): Regularization factor for the current when optimizing. A higher coefficient will
                             penalize higher current values while a lower factor will lower the effect of the
                             regularization. A negative value will favour high currents (not preferred). Only relevant
                             for 'least_squares' opt_method.
-        mask_dilation_kernel (str): kernel used to dilate the mask. Allowed shapes are: 'sphere', 'cross', 'line'
-                                    'cube'. See :func:`shimmingtoolbox.masking.mask_utils.dilate_binary_mask` for more
-                                    details.
-        mask_dilation_kernel_size (int): Length of a side of the 3d kernel to dilate the mask. Must be odd. For example,
-                                         a kernel of size 3 will dilate the mask by 1 pixel.
-        path_output (str): Path to the directory to output figures. Set logging level to debug to output debug
+            path_output (str): Path to the directory to output figures. Set logging level to debug to output debug
                            artefacts.
+        """
+        super().__init__(slices, mask_dilation_kernel, mask_dilation_kernel_size, reg_factor, path_output)
+        self.nii_fieldmap, self.nii_fieldmap_orig, self.extending = self.get_fieldmap(nii_fieldmap)
```

Comment:
```suggestion
        self.nii_fieldmap, self.nii_fieldmap_orig, self.fmap_is_extended = self.get_fieldmap(nii_fieldmap)
```

Change `self.extending` to `self.fmap_is_extended` since just extending is not clear